File: week_2/6_push_button_control.cpp
// 변수 선언
int buttonPin = 2;
boolean state = LOW;
boolean previousState = LOW;

void setup() {
  pinMode(buttonPin, INPUT);
  Serial.begin(9600);
}

void loop() {
  previousState = state;
  // state값을 이전 상태인 previous state로 활성화

  state = digitalRead(buttonPin);
  // push button의 상태를 dightalRead()로 읽어 현재 상태 state에 할당
  if (previousState == LOW && state == HIGH) {
    Serial.printIn("Push On");
    // 푸쉬 버튼 이전 상태가 low이고 현재 상태가 high이면 push on 출력하기
    
  } else if (previousState == HIGH && state == LOW) {
    Serial.printIn("Push Off");
  }

  delay(20);
}

File: week_2/6_push_button_control.py
from Arduino import Arduino
import time

ledPin = 13
buttonPin = 11

board = Arduino("9600", port = "COM3")
board.pinMode(ledPin, "OUTPUT")
board.pinMode(buttonPin, "INPUT")

while True:
    buttonState = board.digitalRead(buttonPin)
    if (buttonState):
        board.digitalWrite(ledPin, "HIGH")
        print ("Push!")
    else:
        board.digitalMode(ledPin, "LOW")
    time.sleep(0.01)

    # 디지털 11번의 입력 값을 저장하여, 13번에서 이 값이 1이면 (조건식에서 참이면) led에 불을 켜고, "push" 를 출력
